node_modules/col-course/lib/api.js

```diff
@@ -55,7 +55,7 @@ var getOrCreateCourse = module.exports.getOrCreateCourse = function(canvasCourse
   DB.Course.findOrCreate(options).complete(function(err, data) {
     if (err) {
       log.error({'err': err}, 'Failed to get or create a course');
-      return callback(err);
+      return callback({'code': 500, 'msg': err.message});
```

Comment:
It's definitely not a widespread convention, although some projects do use it. Its main convenience is that the errors will be easy to deal with once they bubble up to the REST layer and the advantage of speaking HTTP error codes in the error objects is that they usually give quite a bit of information about the nature of the error.